specification/inference/unified_inference/UnifiedRequest.ts

```diff
@@ -0,0 +1,214 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { RequestBase } from '@_types/Base'
+import { Id } from '@_types/common'
+import { Duration } from '@_types/Time'
+import { TaskType } from '@inference/_types/TaskType'
+
+/**
+ * An object style representation of a single portion of a conversation.
+ */
+export interface ContentObject {
+  /**
+   * The text content.
+   */
+  text: string
+  /**
+   * The type of content.
+   */
+  type: string
+}
+
+/**
+ * The function that the model called.
+ */
+export interface ToolCallFunction {
+  /**
+   * The arguments to call the function with in JSON format.
+   */
+  arguments: string
+  /**
+   * The name of the function to call.
+   */
+  name: string
+}
+
+/**
+ * A tool call generated by the model.
+ */
+export interface ToolCall {
+  /**
+   * The identifier of the tool call.
+   */
+  id: string
+  /**
+   * The function that the model called.
+   */
+  function: ToolCallFunction
+  /**
+   * The type of the tool call.
+   */
+  type: string
+}
+
+/**
+ * An object representing part of the conversation.
+ */
+export interface Message {
+  /**
+   * The content of the message.
+   */
+  content: string | Array<ContentObject>
+  /**
+   * The role of the message author.
+   */
+  role: string
+  /**
+   * The tool call that this message is responding to.
+   */
+  tool_call_id?: string
+  /**
+   * The tool calls generated by the model.
+   */
+  tool_calls?: Array<ToolCall>
+}
+
+/**
+ * The tool choice function.
+ *
+ */
+export interface CompletionToolChoiceFunction {
+  /**
+   * The name of the function to call.
+   */
+  name: string
+}
+
+/**
+ * Controls which tool is called by the model.
+ */
+export interface CompletionToolChoice {
+  /**
+   * The type of the tool.
+   */
+  type: string
+  /**
+   * The tool choice function.
+   */
+  function: CompletionToolChoiceFunction
+}
+
+/**
+ * The completion tool function definition.
+ */
+export interface CompletionToolFunction {
+  /**
+   * A description of what the function does.
+   * This is used by the model to choose when and how to call the function.
+   */
+  description?: string
+  /**
+   * The name of the function.
+   */
+  name: string
+  /**
+   * The parameters the functional accepts. This should be formatted as a JSON object.
+   */
+  parameters?: UserDefinedValue
+  /**
+   * Whether to enable schema adherence when generating the function call.
+   */
+  strict?: boolean
+}
+
+/**
+ * A list of tools that the model can call.
+ */
+export interface CompletionTool {
+  /**
+   * The type of tool.
+   */
+  type: string
+  /**
+   * The function definition.
+   */
+  function: CompletionToolFunction
+}
+
+/**
+ * Perform inference on the service using the Unified Schema
+ * @rest_spec_name inference.unified_inference
+ * @availability stack since=8.18.0 stability=stable visibility=public
```

Comment:
@davidkyle Double check that this is what we want

Reply:
This looks good to me.